core/unit_test/TestTeamVector.hpp

```diff
@@ -605,6 +605,71 @@ struct functor_vec_scan {
   }
 };
 
+// Temporary: This condition will progressively be reduced when parallel_scan
+// with return value will be implemented for more backends.
+#if defined(KOKKOS_ENABLE_SERIAL) || defined(KOKKOS_ENABLE_OPENMP)
+#if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_ENABLE_HIP) &&             \
+    !defined(KOKKOS_ENABLE_OPENACC) && !defined(KOKKOS_ENABLE_SYCL) &&         \
+    !defined(KOKKOS_ENABLE_THREADS) && !defined(KOKKOS_ENABLE_OPENMPTARGET) && \
+    !defined(KOKKOS_ENABLE_HPX)
```

Comment:
Why are we not testing if for instance OpenMP+CUDA is enabled?
We can easily do something like
```C++
#ifdef KOKKOS_ENABLE_OPENMP
if constexpr (std::is_same_v<TEST_EXECSPACE, OpenMP>)
#endif
#ifdef KOKKOS_ENABLE_SERIAL
if constexpr (std::is_same_v<TEST_EXECSPACE, Serial>)
#endif
ASSERT_TRUE((TestTeamVector::Test<TEST_EXECSPACE>(12)));

Reply:
I think all the PRs are ready to introduce this for all backends? So it doesn't seem to matter too much?